website/server/controllers/api-v3/groups.js

```diff
@@ -603,7 +603,15 @@ api.joinGroup = {
       group.leader = user._id; // If new user is only member -> set as leader
     }
 
-    group.memberCount += 1;
+    if(group.type === 'party')
```

Comment:
looks like the `lint` command is complaining about the code style, in particular I think the issue is with the lack of space after the if and the bracket not on the same line, something like this should work

```
if (group.type === 'party') {
```

Reply:
can we add a comment explaining why the difference between guilds and parties exist when updating the `memberrCount`? Something like "For parties we count the number of members from the database to get the correct value, see https://github.com/HabitRPG/habitica/issues/12275 on why this is necessary and only done for parties."

Reply:
Added the comment and changed the code style.

Reply:
Since you wanted to replace the custom queries, I used `await group.memberCount()` for both, guilds and parties. Please let me know if we need to keep the original code for guilds or not. 

Which instance related to `memberCount` are you referring to here (/website/server/models/group.js#L1438)? 

Also, I noticed that the lines https://github.com/JalanshMunshi/habitica/blob/6cccb17415ac6037d73460dca83d17f2f96d42f0/website/server/models/group.js#L1422-L1429 and https://github.com/JalanshMunshi/habitica/blob/6cccb17415ac6037d73460dca83d17f2f96d42f0/website/server/models/group.js#L1434-L1441 are same. 

I do not have much experience with Habitica's code, but as per the comments, they are performing the same set of tasks.

Reply:
Good catch, I think the lines going from 1420 to 1429 can be removed